cardano-cli/src/Cardano/CLI/Run/Legacy/Query.hs

```diff
@@ -457,13 +459,16 @@ runQueryKesPeriodInfo socketPath (AnyConsensusModeParams cModeParams) network no
             eInMode <- toEraInMode era cMode
               & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
 
+            shelleyInMode <- toEraInMode ShelleyEra cMode
+              & hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (AnyConsensusMode cMode) anyE)
+
             -- We check that the KES period specified in the operational certificate is correct
             -- based on the KES period defined in the genesis parameters and the current slot number
             eraInMode <- calcEraInMode era $ consensusModeOnly cModeParams
 
             requireNotByronEraInByronMode eraInMode
 
-            gParams <- lift (queryGenesisParameters eInMode sbe)
+            gParams <- lift (queryGenesisParameters shelleyInMode ShelleyBasedEraShelley)
```

Comment:
Even though we are in a particular `era` we can still query for  the genesis parameters which is Shelley specific.

Moreover multiple queries exist over the same connection.

Something worth considering is renaming `queryGenesisParameters` to `queryShelleyGenesisParameters` and taking only the mode as an argument.